Zelda/Zelda.cpp

```diff
@@ -1,1205 +0,0 @@
-}
-
-
-
-void Room::setWest(Room* roomNo)
-{
-	paths[3] = roomNo;
-}
-
-
-
-void Room::setItemsPresent(int itemNo , Item* item)
-{
-	items_Present[itemNo] = item;
-}
-
-
-
-void Room::setMonsterPresent(Monster* monster)
-{
-	monster_Present = monster;
-}
-
-
-
-void Room::setPrincessPresent(Princess* princess)
-{
-	princess_Present = princess;
-}
-
-
-
-void Room::setDesription(char const* room_Description)
-{
-	description = new char [strlen(room_Description)+1];
-	strcpy_s(description,strlen(room_Description)+1,room_Description);
-}
-
-
-
-void Room::setRoomNumber(int roomNo)
-{
-	roomNumber = roomNo;
-}
-
-
-
-//---------------------Castle------------------------
-
-
-
-Castle::Castle()
-{
-
-}
-
-
-
-Room* Castle::getRoom( int room_No )
-{
-	return &rooms[room_No - 1];
-}
-
-
-
-void Castle::LinkRoomsWithOtherThings(Item** items , Monster** monsters , Princess* princess)
-{
-	//Room1
-	rooms[0].setSouth(&rooms[3]);
-	rooms[0].setEast(&rooms[1]);
-	//room1.west = Exit from castle
-
-
-	//Room2
-	rooms[1].setWest(&rooms[0]);
-	rooms[1].setEast(&rooms[2]);
-	rooms[1].setSouth(&rooms[4]);
-
-	rooms[1].setItemsPresent(0 , items[0]);
-
-
-	//Room3
-	rooms[2].setWest(&rooms[1]);
-	rooms[2].setItemsPresent(0 , items[3]);
-
-
-	//Room4
-	rooms[3].setNorth(&rooms[0]);
-	rooms[3].setItemsPresent(0 , items[1]);
-
-
-	//Room5
-	rooms[4].setNorth(&rooms[1]);
-	rooms[4].setEast(&rooms[5]);
-
-	rooms[4].setMonsterPresent( monsters[0] );
-
-
-	//Room6
-	rooms[5].setWest(&rooms[4]);
-
-	rooms[5].setMonsterPresent( monsters[1] );
-
-
-	//Room7
-	rooms[6].setEast(&rooms[7]);
-	rooms[6].setItemsPresent(0 , items[4]);
-
-
-	//Room8
-	rooms[7].setWest(&rooms[6]);
-	rooms[7].setNorth(&rooms[4]);
-
-	rooms[7].setItemsPresent(0 , items[2]);
-
-
-	//Room9
-	rooms[8].setNorth(&rooms[5]);
-	rooms[8].setPrincessPresent(princess);
-}
-
-
-
-void Castle::LinkRoom5and8()
-{
-	rooms[4].setSouth(&rooms[7]);
-	rooms[7].setNorth(&rooms[4]);
-}
-
-
-
-void Castle::LinkRoom6and9()
-{
-	rooms[5].setSouth(&rooms[8]);
-	rooms[8].setNorth(&rooms[5]);
-}
-
-
-
-void Castle::HiddenRoomsUnlocker(char const* monsterName , Monster** monsters)
-{
-	if (strcmp(monsterName , (monsters[0]->getMonsterName())) == 0)
-	{
-		LinkRoom5and8();
-	}
-
-	else if (strcmp(monsterName , (monsters[1]->getMonsterName())) == 0)
-	{
-		LinkRoom6and9();
-	}
-}
-
-
-
-void Castle::setDescriptionOfRooms()
-{
-	ifstream InputStream("Rooms.txt");
-	if (InputStream.is_open())
-	{
-		char buffer[300];
-
-		for (int i = 0; i < 9 && !InputStream.eof(); ++i)
-		{
-			InputStream.getline(buffer,300);
-			rooms[i].setDesription(buffer);
-		}
-	}
-	else
-		cout << "Could not open file";
-	InputStream.close();
-}
-
-
-
-void Castle::setNumbersofRooms()
-{
-	for (int i = 0; i < 9; ++i)
-	{
-		rooms[i].setRoomNumber(i+1);
-	}
-}
-
-
-
-Castle::~Castle()
-{
-
-}
-
-
-
-//-------------------Player---------------------
-
-
-
-Player::Player(char const* player_Name , Room* room1 )
-{
-	name = new char [strlen(player_Name) + 1];
```

Comment:
Why does this have to be dynamically allocated?  The size of the name is at most 32 bytes.  Use a static array instead.